openraft/src/core/raft_core.rs

```diff
@@ -1197,6 +1136,79 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> RaftCore<C,
         Ok(())
     }
 
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn handle_tick_election(&mut self) -> Result<(), StorageError<C::NodeId>> {
+        let now = *self.engine.timer.now();
+
+        tracing::debug!("try to trigger election by tick, now: {:?}", now);
+
+        // TODO: leader lease should be extended. Or it has to examine if it is leader
+        //       before electing.
+        if self.engine.state.server_state == ServerState::Leader {
```

Comment:
@drmingdrmer do we skip for Candidate state also here? I don't know what if a candidate handles tick and runs the following logic to re-elect itself.